metadata/src/main/java/org/apache/kafka/controller/ClusterControlManager.java

```diff
@@ -327,6 +328,9 @@ public ControllerResult<BrokerRegistrationReply> registerBroker(
         }
         int brokerId = request.brokerId();
         BrokerRegistration existing = brokerRegistrations.get(brokerId);
+        if (version < 2 || request.previousBrokerEpoch() != existing.epoch()) {
+            log.debug("Received an unclean shutdown request");
```

Comment:
Could we add a TODO to indicate that the implementation is not complete?

Reply:
Done